### ISSUE TITLE
Bug: `sort-modules` by `usage` doesn't seem to work

### ISSUE THREAD

**Issue by aguo-ntech** [azat-io/eslint-plugin-perfectionist#666]
### Describe the bug

The new `usage` option for `sort-modules` (introduced by @hugop95 in #648) i documented as:

> Sort items by the order they are first used or referenced within their [group](https://perfectionist.dev/rules/sort-modules#groups).

Therefore, in this code example, I would expect that `A` would be sorted before `B`, since `A` is used/referenced before `B`. However, the plugin does not report any issues, and is happy with the current sorting order.

### Code example

```
type B = 1;

type A = 2;

type X = A | B;

### ESLint version

9.39.2

### ESLint Plugin Perfectionist version

5.2.0

### Additional comments

_No response_

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @aguo-ntech 

The current behavior is intended: `usage` enforces that a module element A (function, type, interface, class, enum) is placed before another module element B if B references A. However, it doesn't enforce anything else.

This sort option is necessarily opinionated, and in my opinion, enforcing what you suggest feels like too much, we should at least give the possibility to users to enforce lenient usage sorting while still giving them the freedom to sort first-used elements as they want (alphabetically, for example).

We can first work on documenting the option better to avoid confusion, and in a second time think about upgrading the option to support stricter enforcement.

**Comment by aguo-ntech** [comment]
Ah.

In that case, is it correct that this `usage` option is identical to the `no-use-before-define` rule from [ESLint](https://eslint.org/docs/latest/rules/no-use-before-define) and [typescript-eslint](https://typescript-eslint.io/rules/no-use-before-define/)?

> This sort option is necessarily opinionated, and in my opinion, enforcing what you suggest feels like too much, we should at least give the possibility to users to enforce lenient usage sorting while still giving them the freedom to sort first-used elements as they want (alphabetically, for example).

> in a second time think about upgrading the option to support stricter enforcement.

I definitely think that this would be a good idea.

Everything else in `eslint-plugin-perfectionist` has a strict and specific sort order — this entire ESLint plugin is about being very opinionated — so I was very surprised that this `usage` option is _not_ opinionated.

**Comment by hugop95** [comment]
@aguo-ntech 

> In that case, is it correct that this usage option is identical to the no-use-before-define rule from [ESLint](https://eslint.org/docs/latest/rules/no-use-before-define) and [typescript-eslint](https://typescript-eslint.io/rules/no-use-before-define/)?

This is interesting, I didn't think about this rule. I think that it does look similar indeed. The main value of `usage` in comparison should come with `fallbackSort`, for users who want to enforce what `no-use-before-define` does while also sorting `alphabetically`.

I realize that we don't support `fallbackSort` correctly though for this use case, this is a bug that will be fixed.

--- 

> Everything else in eslint-plugin-perfectionist has a strict and specific sort order — this entire ESLint plugin is about being very opinionated — so I was very surprised that this usage option is not opinionated.

Honestly, I don't think that `perfectionist` is that opinionated overall: we offer opinionated default options that users can always override with a good granularity, and I think that we have reached a state where users can generally achieve what they have in mind by playing with the offered options. That's not the case for `usage` though, as the option is new and we don't really know how users want to use it. 

> so I was very surprised that this usage option is not opinionated.

Opinionating/enforcing things too much right away has a cost: loosening a strong enforcement with `eslint-disable` is more burdensome than enforcing less and manually adjusting things if needed.

**Comment by azat-io** [comment]
Fixed in a2b898e4d296ab24cd2ff0c107cd0aa96b174e02.
Released in [v5.4.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v5.4.0).
